libuserhelper.cpp

```diff
@@ -0,0 +1,255 @@
+        qWarning() << "Error creating context:" << lu_strerror(error);
+        lu_error_free(&error);
+        return false;
+    }
+
+    bool rv = true;
+    struct lu_ent *ent = lu_ent_new();
+    struct lu_ent *entGroup = lu_ent_new();
+
+    if (lu_user_lookup_id(context, uid, ent, &error)) {
+        const char *user = lu_ent_get_first_string(ent, LU_USERNAME);
+        // Forced to use deprecated GValueArray that libuser is using
+        GValueArray *groups = lu_groups_enumerate_by_user(context, user, &error);
+        if (groups) {
+            GValue *value = g_slice_new0(GValue);
+            g_value_init(value, G_TYPE_STRING);
+            g_value_set_string(value, user);
+
+            // Remove user from groups
+            for (uint i = 0; i < groups->n_values; i++) {
+                if (lu_group_lookup_name(context, g_value_get_string(g_value_array_get_nth(groups, i)), entGroup, &error)) {
+                    lu_ent_del(entGroup, LU_MEMBERNAME, value);
+                    if (!lu_group_modify(context, entGroup, &error)) {
+                        qWarning() << "Group modify failed:" << lu_strerror(error);
+                        lu_error_free(&error);
+                        rv = false;
+                        break;
+                    }
+                }
+            }
+        } else {
+            lu_error_free(&error);
+            error = nullptr;
+        }
+
+        if (rv && lu_group_lookup_id(context, lu_ent_get_first_id(ent, LU_GIDNUMBER), entGroup, &error)) {
+            if (lu_group_delete(context, entGroup, &error)) {
+                if (!lu_user_delete(context, ent, &error)) {
+                    qWarning() << "User delete failed:" << lu_strerror(error);
+                    lu_error_free(&error);
+                    rv = false;
+                }
+            } else {
+                qWarning() << "Group delete failed:" << lu_strerror(error);
+                lu_error_free(&error);
+                rv = false;
+            }
+        } else {
+            qWarning() << "Could not find group";
+            rv = false;
+        }
+    } else {
+        qWarning() << "Could not find user";
+        rv = false;
+    }
+
+    lu_ent_free(ent);
+    lu_ent_free(entGroup);
+    lu_end(context);
+
+    return rv;
+}
+
+bool LibUserHelper::modifyUser(uint uid, const QString &new_user, const QString &new_name)
+{
+    struct lu_error *error = nullptr;
+    struct lu_context *context = lu_start(NULL, lu_user, NULL, NULL, lu_prompt_console, NULL, &error);
+    if (!context) {
+        qWarning() << "Error creating context:" << lu_strerror(error);
+        lu_error_free(&error);
+        return false;
+    }
+
+    bool rv = true;
+    struct lu_ent *ent = lu_ent_new();
+    if (lu_user_lookup_id(context, uid, ent, &error)) {
+        lu_ent_set_string(ent, LU_USERNAME, new_user.toUtf8());
+        QString fixedName(new_name);
+        lu_ent_set_string(ent, LU_GECOS, fixedName.remove(',').toUtf8());
```

Comment:
This should sanitize for ':' as well, unless libuser does that and we rely on that.